kin-sdk/kin-sdk-lib/src/test/kotlin/kin/sdk/internal/queue/FakeQueueScheduler.kt

```diff
@@ -0,0 +1,57 @@
+package kin.sdk.internal.queue
+
+import java.util.*
+import kotlin.concurrent.schedule
+
+open class FakeQueueScheduler : QueueScheduler {
+
+    private var timer: Timer? = Timer()
+    private val futureTasks: HashMap<Runnable?, TimerTask?> = HashMap()
+
+    //TODO what is better? one of the commented one or the non commented one. Which one?
```

Comment:
I dealt with the bg of the timer in a naive way, meaning for example that a task should start in 2 seconds than to be on the safe side I did the assertions only after 3 seconds.
Currently, it is working but in order to do it like in production code then we probably need listeners for when the task is completed or use a latch or something like that and I wasn't sure if it is really necessary here.
what do you say?

Regarding your first comment, there are 3 uncommented methods. One with Handler which is probably not good because it is Android.
2 others with timer.
The only question is what is the difference between those 3 methods with the timer?
something in Kotlin probably.